### DISCUSSION TITLE
Export utility types/functions through @kadena/client?

### DISCUSSION THREAD

**Discussion by alber70g** [kadena-community/kadena.js#72] [Ideas]
I ran in my testproject, [transfer.ts](https://github.com/kadena-community/kadena.js/blob/c17da4316c63abf95bf5b5dbd3bbe718c5092261/packages/libs/pactjs-test-project/src/transfer.ts#L11) into the PactNumber. I see two options

1. the consuming library imports `PactNumber` from `@kadena/pactjs`
2. we re-export `PactNumber` in `@kadena/client` and the consuming library imports from `@kadena/client`

**1.** we have less control on the versions that users use. They can mix and match different versions together. The consumer needs to make sure to use compatible versions. We can provide `peerDependencies` in `@kadena/client` for `@kadena/pactjs` (e.g. `@kadena/pactjs@>=2.8.0 || >= 3.2.0-dev || >= 3.3.0-dev`) and other libraries.

**2.** we decide what version of `@kadena/pactjs` (or other dependencies) to use and export ourselves in `index.ts` `export * from '@kadena/pactjs'`. The consumer has now **one** library to use. They cannot mix and match other versions. 
For us this is easier to maintain, because our tests will validate if the combination used will work. No issues in the community of "which version works where". We can push refactorings easier when we expose one "entry library" (i.e. `@kadena/client`)

**Comment by Randynamic** [comment]
My preference goes out to 1 lib overall, like nr. 2.
Naming I would instead use `@kadena/kadenajs` as the client will be a part of the libs kadenajs will expose.
